test/mongodb.test.js

```diff
@@ -547,6 +687,200 @@ describe('mongodb connector', function() {
+              'unitprice': '45.50',
+            },
+          ],
+        };
+        modelWithDecimalNestedArray.updateAll({id: inst.id}, updateData, function(err, inst) {
+          inst.toObject().tickets[0].unitprice.should.equal(Decimal128.fromString('27.5'));
+        });
+        done();
+      });
+    });
+
+    it('should create/update instance for nested decimal prop inside object', function(done) {
+      modelWithDecimalNestedObject.create({
+        'awards': {
+          'currency': 'USD',
+          'wins': '4',
+          'prizeMoney': '100000.00',
+        },
+      }, function(err, inst) {
+        inst.toObject().awards.prizeMoney.should.equal(Decimal128.fromString('10000.00'));
+        var updateData = {
+          'awards': {
+            'currency': 'CAD',
+            'wins': '10',
+            'prizeMoney': '25000.00',
+          },
+        };
+        modelWithDecimalNestedObject.updateAll({id: inst.id}, updateData, function(err, inst) {
+          inst.toObject().awards.prizeMoney.should.equal(Decimal128.fromString('25000.00'));
+        });
+        done();
+      });
+    });
+    it('should create/update instance for deeply nested decimal props', function(done) {
+      modelWithDeepNestedDecimalProps.create({
+        'imdb': {
+          'reviewDate': '2019-01-29T04:00:39.828Z',
+          'innerArray': [
+            {
+              'ObjInsideAnArray': {
+                'testNumber': '1234',
+                'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                'testDecimal': '99.55',
+              },
+              'nestedArray': [
+                {
+                  'testNumber': '1234',
+                  'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                  'testDecimal': '99.55',
+                },
+                {
+                  'testNumber': '1234',
+                  'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+                  'testDecimal': '99.55',
+                },
+              ],
+              'testNumber': '1234',
+              'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+              'testDecimal': '99.55',
+            },
+            {
+              'ObjInsideAnArray': null,
+              'nestedArray': null,
+              'testNumber': '5678',
+              'testDate': "TIMESTAMP '2019-01-30 18:26:38.551022'",
+              'testDecimal': '99.55',
+            },
+          ],
+          'innerObj': {
+            'innerObj': {
+              'innerObj': {
+                'innerObj': {
+                  'testNumber': '6666',
+                  'testDate': '2019-01-30 18:26:38.551022',
+                  'testDecimal': '55.55',
+                },
+              },
+            },
+          },
+          'duration': '135',
+          'rating': '4.5',
+        },
+      }, function(err, inst) {
+        // inst.toObject().awards.prizeMoney.should.equal(Decimal128.fromString('25000.00'));
```

Comment:
any reason to comment this assertion?

Reply:
There is a bug in juggler right now where these assertions will fail because juggler doesn't return the updated model instance on create/update :(. Even though they are stored as decimal128 in mongo, we get back string values. I aim to fix that in juggler, but I was manually able to verify that they are stored correctly.

Reply:
> because juggler doesn't return the updated model instance on create/update

I think that's an intentional behavior, see https://loopback.io/doc/en/lb3/Operation-hooks.html

> By default, create and updateAttributes do not apply database updates to the model instance returned to the callback, therefore any changes made by “loaded” hooks are discarded. To change this behavior, set a per-model option `updateOnLoad: true`.

I think it's more important to verify what has been actually stored by the database. You can start by calling `findById` to fetch the actual data. IDK if `decimal128` type is preserved on that path. If not, then you may need to execute custom MongoDB command to access raw data.

Reply:
`findById` did not work, but yeah perhaps using custom MongoDB command would work here 👍